chromium_src/media/audio/pulse/pulse_util.cc

```diff
@@ -3,7 +3,48 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this file,
 // you can obtain one at http://mozilla.org/MPL/2.0/.
 
-constexpr char kBrowserDisplayName[] = "brave-browser";
+#include "base/strings/string_util.h"
+#include "build/build_config.h"
+
+#define STABLE_DISPLAY_NAME "brave-browser"
+#define DEV_DISPLAY_NAME "brave-browser-dev"
+#define BETA_DISPLAY_NAME "brave-browser-beta"
+#define NIGHTLY_DISPLAY_NAME "brave-browser-nightly"
+
+namespace media {
+
+namespace pulse {
+
+namespace {
+    // We can't use brave::GetChannelName() because audio is a dependency of
+    // common, so we implement minimal channel checking here to determine which
+    // icon to use. This is Linux only so it's fine just using those modifiers.
+    //
+    // The defines are pulled in via a very small patch to the audio BUILD.gn.
+    const char* BrowserIconName() {
```

Comment:
Is there a reason why we'd want to do this at runtime instead of the build time?

I wonder if we could just define `LINUX_CHANNEL=\"$linux_channel\"` in `//brave/build/linux:linux_channel_names` and then use it to define display name as a combination of "brave-browser-" and LINUX_CHANNEL (gated on OFFICIAL_BUILD as well).

Also, I am not sure if this would work, but maybe we can redefine `pa_proplist_sets` to our own version in this override to avoid patching pulse_util.cc, or redefining PA_PROP_APPLICATION_ICON_NAME into 2 params and having our own overload for pa_proplist_sets - not sure if either would work, but it's worth a try to avoid patching.